File: alumno.java
public class alumno {
    
    String nombre;
    String curso;
    double promedio;

    public alumno(String nombre, String curso, double promedioInicial) {
        this.nombre = nombre;
        this.curso = curso;
        this.promedio = promedioInicial;
    }

    public void saludar() {
        System.out.println("Hola, soy " + this.nombre + " y estoy en el curso " + this.curso);
    }
    
    public void mostrarPromedio() {
        System.out.println("El promedio actual de " + this.nombre + " es: " + this.promedio);
    }
    
    public void aumentarPromedio(double valor) {
        this.promedio += valor; 
        System.out.println(this.nombre + " ha aumentado su promedio en " + valor + ".");
        this.mostrarPromedio();
    }
}

File: trabajo 4/alumno.js
class Alumno {
    
    constructor(nombre, curso, promedioInicial) {
        
        this.nombre = nombre;
        this.curso = curso;
        this.promedio = promedioInicial;
    }

    saludar() {
        console.log(`Hola, soy ${this.nombre} y estoy en el curso ${this.curso}.`);
    }

    mostrarPromedio() {
        console.log(`El promedio actual de ${this.nombre} es: ${this.promedio}`);
    }

    aumentarPromedio(valor) {
        
        this.promedio += valor;
        console.log(`${this.nombre} ha aumentado su promedio en ${valor} puntos.`);
        this.mostrarPromedio(); 
    }
}
